### PR TITLE
Fix sibyll rng

### PR DESCRIPTION
**Pull request by jncots**
Here are fixes for Sibyll models.

* `Sibyll2.3d`: There were a problem with wrong change of rng function GASDEV to SIMRND. It is reverted now. GASDEV uses SIMRND and has internal state ISET which is introduced to use both values drawn from Gaussian distribution. To be able a restoration of state, `RMMARDSib` subclass of `RMMARDState` is added to `sibyll.py`, which contains the additional state variable `_gasdev_iset` obtained from `RNDMGAS` common block.
* `Sibyll2.1`: In addition to problems of `Sibyll2.3d` the following problem places were fixed:
    * SPGASDEV is used instead of  GASDEV, and there could be a conflict of `RNDMGAS` common blocks of both functions. Fix:  At a build time of  `Sibyll2.1`  `GASDEV` is removed by preprocessor
    * Real version of `S_RNDM()` can produce 1.0, because of casting to real. Fix: regenerate next random number, if 1.0 is produced.
    * `SIBYLL` function at a start can be in a different state because of `PPT0` could be changed during run. This results in different state of a generator, even if state of rng is set to correct state. Fix: set `PPT0` to the values given in block data at every call to function `SIBYLL`.

### CODE REVIEW

**Comment by afedynitch** [comment]
Yes, it fixes the issue. I forgot the exact reason why it could pass the previous tests despite Anton's explanation. But something like this should not happen anymore with the random number state machinery in place. We can check directly if the test events reproduced exactly the same results.

In this PR only one small issue remaining in how SIBYLL2.1 was fixed that should be done in a different way, but the rest is straight forward. So tomorrow it will be merged.

**Comment by jncots** [comment]
There was an issue that PPT0 was assigned to hardcoded values. It has now been modified to restore the values specified in the block data.

**Review by afedynitch — APPROVED** [review]
Please put your own initials into the Fortran code before merging not mine. In case it will be important for something in future, my Fortran initials are “anfe”.

File: src/fortran/sibyll/sibyll_21.f [hunk 1]
```diff
@@ -104,6 +104,18 @@ SUBROUTINE SIBYLL (K_beam, IATARG, Ecm)
...
       NSOF = 0
       IATARGET = IATARG
+      
+C...afed 06.10.2022      
+C...Set PPT0 back to initial values given in block data:
+C...DATA PPT0 /0.30,0.30,0.450,30*0.60/
+C...When ptflag=1 (as in default), PPT0 is different
+C...from initial values - it is a potential PROBLEM!!!
+      PPT0(1) = 0.30
+      PPT0(2) = 0.30
```
[the remark this answers is not in the document]
Comment (afedynitch) [reply src/fortran/sibyll/sibyll_21.f]: Solved by latest commit.

File: src/fortran/sibyll/sibyll_21.f [hunk 2]
```diff
@@ -104,6 +104,18 @@ SUBROUTINE SIBYLL (K_beam, IATARG, Ecm)
       NJET = 0
       NSOF = 0
       IATARGET = IATARG
+      
+C...afed 06.10.2022
```
[the remark this answers is not in the document]
Comment (jncots) [reply src/fortran/sibyll/sibyll_21.f]: Ok, it is changed to my initials.

File: src/fortran/sibyll/sibyll_21.f (hunk 1 above)
[the remark this answers is not in the document]
Comment (jncots) [reply src/fortran/sibyll/sibyll_21.f]: Done in [1badcc6](https://github.com/impy-project/impy/pull/73/commits/1badcc62f9b8dee2497deb58019b093ec0841bd9).

**Pull request merged**